### PR TITLE
Drop support for Python 3.7 and Pyspark 2.x and remove vendored dependency on six

### PR DESCRIPTION
**Pull request by fpgmaas**
This PR:
- Removes the dependency on `six`.
- Drops support for Python 3.7, since it is now [EOL for over a year](https://devguide.python.org/versions/).
- Updates some development dependencies related to pytest, so the unit tests can be run on newer versions of Python as well.
- Adds the `.python_version` to `.gitgnore`.

For removing the dependency on `six`, I simply replaced `six.moves.zip_longest` with `itertools.zip_longest` as suggested by @MrPowers. I however did not test if this changes existing functionality, I assume that is already covered by the unit tests? Those ran uccessfully.

See also https://github.com/MrPowers/chispa/issues/78

### CODE REVIEW

**Comment by fpgmaas** [comment]
@MrPowers I think this one is ready for review; is it time to drop support for Python 3.7? If not I can reset it to `>=3.7` and we'll leave that for another PR.

**Comment by fpgmaas** [comment]
@SemyonSinchenko Do you have time to review this one by any chance?

**Comment by SemyonSinchenko** [comment]
@fpgmaas I can, of course, but I do not have write access here.
@MrPowers

File: pyproject.toml [hunk 1]
```diff
@@ -6,13 +6,13 @@ authors = ["MrPowers <matthewkevinpowers@gmail.com>"]
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = ">=3.7,<4.0"
+python = ">=3.8,<4.0"
```
Comment (SemyonSinchenko) [inline pyproject.toml:9]: 3.8 is EOL very soon too, let's drop it?

Comment (fpgmaas) [reply pyproject.toml:9]: Good point, however I think it's good to keep supporting it until it's officially EOL, unless something is really blocking us from improving this project further. Are there any issues, e.g. spark compatibility issues?

Comment (SemyonSinchenko) [reply pyproject.toml:9]: No, no known issues. Ok, let's wait until official EOL!

**Review by SemyonSinchenko — APPROVED** [review]
LGTM

**Pull request merged**